tensorzero-core/src/optimization/gcp_vertex_gemini_sft/mod.rs

```diff
@@ -0,0 +1,462 @@
+                epoch_count: self.n_epochs,
+                adapter_size: self.adapter_size,
+                learning_rate_multiplier: self.learning_rate_multiplier,
+            }),
+            export_last_checkpoint_only: self.export_last_checkpoint_only,
+        };
+
+        let encryption_spec = EncryptionSpec {
+            kms_key_name: self.kms_key_name.clone(),
+        };
+
+        let body = GCPVertexGeminiFineTuningRequest {
+            base_model: self.model.clone(),
+            supervised_tuning_spec,
+            tuned_model_display_name: self.tuned_model_display_name.clone(),
+            service_account: self.service_account.clone(),
+            encryption_spec: Some(encryption_spec),
+        };
+
+        let url = gcp_vertex_gemini_url(&self.project_id, &self.region, None).map_err(|e| {
+            Error::new(ErrorDetails::InvalidBaseUrl {
+                message: e.to_string(),
+            })
+        })?;
+
+        let auth_headers = self
+            .credentials
+            .get_auth_headers(
+                &format!("https://{}-aiplatform.googleapis.com/", self.region),
+                credentials,
+            )
+            .await?;
+
+        let request = client.post(url).headers(auth_headers);
+        let res = request.json(&body).send().await.map_err(|e| {
+            Error::new(ErrorDetails::InferenceClient {
+                status_code: e.status(),
+                message: format!(
+                    "Error sending request to OpenAI: {}",
+                    DisplayOrDebugGateway::new(e)
+                ),
+                provider_type: PROVIDER_TYPE.to_string(),
+                raw_request: Some(serde_json::to_string(&body).unwrap_or_default()),
+                raw_response: None,
+            })
+        })?;
+
+        let raw_response = res.text().await.map_err(|e| {
+            Error::new(ErrorDetails::InferenceClient {
+                status_code: e.status(),
+                message: format!(
+                    "Error reading response from GCP Vertex Gemini: {}",
+                    DisplayOrDebugGateway::new(e)
+                ),
+                provider_type: PROVIDER_TYPE.to_string(),
+                raw_request: Some(serde_json::to_string(&body).unwrap_or_default()),
+                raw_response: None,
+            })
+        })?;
+        let job: GCPVertexGeminiFineTuningJob =
+            serde_json::from_str(&raw_response).map_err(|e| {
+                Error::new(ErrorDetails::InferenceServer {
+                    message: format!(
+                        "Error parsing JSON response: {}",
+                        DisplayOrDebugGateway::new(e)
+                    ),
+                    raw_request: Some(serde_json::to_string(&body).unwrap_or_default()),
+                    raw_response: Some(raw_response.clone()),
+                    provider_type: PROVIDER_TYPE.to_string(),
+                })
+            })?;
+        let job_id = job
```

Comment:
From what I can tell, `job_id` is only ever used to reconstruct the job url. I think we could instead just append `job.name` to the base url (e..g `https://{region}-aiplatform.googleapis.com/v1/{job_name}`) to construct `job_url`, and only store `job_url` in `GCPVertexGeminiSFTJobHandle`